ebodac/src/main/java/org/motechproject/ebodac/repository/VisitDataService.java

```diff
@@ -36,6 +37,11 @@
     @Lookup
     List<Visit> findVisitsByType(@LookupField(name = "type") VisitType visitType);
 
+    @Lookup
+    List<Visit> findVisitsByTypeAndDate(@LookupField(name = "type") VisitType visitType,
```

Comment:
I think that you need to check if Prime Vaccination Date in the Participant data is null